src/main/java/Task/TaskList.java

```diff
@@ -0,0 +1,297 @@
+        StringBuilder sb = new StringBuilder();
+        sb.append(task.getTaskTypeAsString()).append(" | ");
+        sb.append(task.getStatus().equals("X") ? "1" : "0").append(" | ");
+        sb.append(task.readTask());
+        if (task instanceof Deadline) {
+            sb.append(" | ").append(((Deadline) task).getBy());
+        } else if (task instanceof Event) {
+            sb.append(" | ").append(((Event) task).getStart());
+            sb.append(" | ").append(((Event) task).getEnd());
+        }
+        return sb.toString();
+    }
+
+    private void saveTasks() {
+        File file = new File(save.getPath());
+
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(file))) {
+            for (Task task : taskList) {
+                writer.write(formatTaskForFile(task));
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            System.out.println("Error saving tasks to file: " + e.getMessage());
+        }
+    }
+
+    private void loadTasksFromSave() {
+        File file = new File(save.getPath());
+        if (!file.exists()) {
+//            System.out.println("File does not exist");
+            return;
+        }
+
+        try (Scanner scanner = new Scanner(file)) {
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                System.out.println(line);
+                Task task = parseTask(line);
+                taskList.add(task);
+                taskListLength++;
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("Error loading tasks from file: " + e.getMessage());
+        }
+    }
+
+
+    private Task parseTask(String line) {
+        String[] parts = line.split(" \\| ");
+        Task.TaskType taskType = Task.TaskType.valueOf(parts[0]);
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+        Task task;
+        list();
+        switch (taskType) {
+        case T:
+            task = ToDo.of(description, taskType);
+            saveTasks();
+            break;
+        case D:
+            try {
+                task = Deadline.of(description + " /by " + parts[3], taskType);
+                saveTasks();
+            } catch (TaskCreationException e) {
+                throw new IllegalArgumentException("Invalid task type: " + taskType);
+            }
+            break;
+        case E:
+            try {
+                task = Event.of(description + "/from " + parts[3] + " /to " + parts[4], taskType);
+                saveTasks();
+            } catch (TaskCreationException e) {
+                System.out.println("Error occurred while parsing task");
+                throw new IllegalArgumentException("Invalid task type: " + taskType);
+            }
+            break;
+        default:
+            throw new IllegalArgumentException("Invalid task type: " + taskType);
+        }
```

Comment:
Nice use of switch statements here! Great job in following the coding standard for the module too with regards to indentation for the different cases!